### ISSUE TITLE
class << self is ignored in Metrics/ModuleLength

### ISSUE THREAD

**Issue by d4rky-pl** [rubocop/rubocop#12231]
## Expected behavior

The code inside `class << self` should be counted towards module length

## Actual behavior

The contents of the block are ignored and `Metrics/ModuleLength` passes

## Steps to reproduce the problem

```yaml
# .rubocop.yml
AllCops:
  DisabledByDefault: true

Metrics/ModuleLength:
  Max: 1
```

```ruby
# This will fail with C: Metrics/ModuleLength: Module has too many lines. [2/1]
module TestModule
  def a
  end
end
```

```ruby
# This will pass even if we put 1000+ more lines in class << self
module TestModule
  class << self
    def a
    end
  end
end
```

## RuboCop version

```
➜  rubocop-repro rubocop -V
1.56.4 (using Parser 3.2.2.3, rubocop-ast 1.29.0, running on ruby 3.1.3) [arm64-darwin22]
```

**Comment by koic** [comment]
This is detected by `Metrics/ClassLength` cop, not `Metrics/ModuleLength` cop.

**Comment by d4rky-pl** [comment]
@koic I'm not sure if this is correct behaviour, if it is then feel free to close this ticket:

Given this configuration:
```yml
AllCops:
  DisabledByDefault: true

Metrics/ModuleLength:
  Max: 1

Metrics/ClassLength:
  Max: 1
```

For this case:

```ruby
module TestModule
  class << self
    def a
    end
  end
end
```

we'll get:
```
test3.rb:3:3: C: Metrics/ClassLength: Class has too many lines. [2/1]
  class << self ...
  ^^^^^^^^^^^^^
```

but for this case:

```ruby
module TestModule
  class << self
    def a
    end
  end
  def a; end
end
```

we'll get both:

```
test3.rb:2:1: C: Metrics/ModuleLength: Module has too many lines. [5/1]
module TestModule ...
^^^^^^^^^^^^^^^^^
test3.rb:3:3: C: Metrics/ClassLength: Class has too many lines. [2/1]
  class << self ...
  ^^^^^^^^^^^^^
```

**Comment by d4rky-pl** [comment]
There was a change in behaviour between 1.49.0 and 1.50.0:

```ruby
# testfile
module TestModule
  class << self
    def a
    end
  end
end
```

there is a difference in which cop reacts:

```
# rubocop 1.49.0

test3.rb:2:1: C: Metrics/ModuleLength: Module has too many lines. [4/1]
module TestModule ...
^^^^^^^^^^^^^^^^^
```

```
# rubocop 1.50.0

test3.rb:3:3: C: Metrics/ClassLength: Class has too many lines. [2/1]
  class << self ...
  ^^^^^^^^^^^^^
```

I assume it's because of either #11773 or #11776 so at this point I'm just not sure if both cops shouldn't react in this case or if the current behaviour is the correct and expected one

**Comment by koic** [comment]
You're right. This is an unexpected change in behavior. I've opened #12240 to resolve this issue. Thank you!
